components/cam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -763,8 +763,13 @@ subroutine get_cloud_fraction(its,ite,kts,kte,ast,qc,qr,qitot,method, &
           rcldm(i,k) = cldm(i,k)
           IF (trim(method) == 'in_cloud') THEN
              IF (k /= ktop) THEN
+                ! If cloud mass exists, keep precip_frac = cldm (precipitation
+                ! only present in cloud). If not, use the max of cloud fraction
+                ! and fraction from the level above (precip is originating from
+                ! above in this case, but we need to use the max because P3 can't
+                ! handle rcldm < cldm correctly).
```

Comment:
I suggest changing this comment to: "in_cloud means that precip_frac = cloud frac when cloud mass is present. Below cloud, precip frac is equal to the cloud fraction from the last layer that had cloud. Since presence or absence of cloud is defined as mass > qsmall, sub-cloud precip frac for the in_cloud method tends to be very small and is very sensitive to tiny changes in condensate near cloud base."

Reply:
for sure, added the comment